### ISSUE TITLE
Attach latest console errors

### ISSUE THREAD

**Issue by akrigline** [League-of-Foundry-Developers/bug-reporter#61]
https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror

This looks like it would let us locally keep track of errors and attach them to reports.

**Comment by Ethck** [comment]
window.onerror() and adding an event listener for errors do not seem to catch ANY errors outside physically calling window.onerror... I'll keep looking.

**Comment by Ethck** [comment]
In regards to error handling, it doesn't look like we can do much without some extensive patching. Basically the main problem is that `window.onerror()` callbacks are only called when there is no error handler in place already. This means that if a dev actually uses try and catch blocks, then we don't receive the errors. There are also numerous problems with sync/async execution and calling `window.onerror()`. I'm not saying it's impossible, but it appears to be fairly intensive work (and way beyond my 4 month old knowledge of JS). At least, that's my understanding.

**Comment by IamWarHead** [comment]
hey there, just getting back with the result of my investigation about collecting errors
unfortunately we wont be able to catch errors which are in a try / catch block or in a promise if rejection is handled. we should provide a wrapper function similar to try/catch which would catch the bugs and it could do whatever with them. without that it will be hard or nearly impossible. browser's console can do it because errors are handled on a lower level.
if someone found any other possible solution please let me know because i am very interested in it

**Comment by typhonrt** [comment]
Left a message in the Leagues Discord which some of you saw. I remember struggling to handle uncaught errors / promise rejections in initial prototyping of a catch all error handler. I'm sure I tried onError first and perhaps a few other things, but they didn't work.. This did work, so give it a try:
```
window.addEventListener('error', event => { s_ERROR_HANDLER(event.error, event); });
window.addEventListener('unhandledrejection', event => { s_ERROR_HANDLER(event.reason, event); });
```

**Comment by IamWarHead** [comment]
Im

> Left a message in the Leagues Discord which some of you saw. I remember struggling to handle uncaught errors / promise rejections in initial prototyping of a catch all error handler. I'm sure I tried onError first and perhaps a few other things, but they didn't work.. This did work, so give it a try:
> 
> ```
> window.addEventListener('error', event => { s_ERROR_HANDLER(event.error, event); });
> window.addEventListener('unhandledrejection', event => { s_ERROR_HANDLER(event.reason, event); });
> ```

Im affraid it wont solve the issue if someone using a `try...catch` block in the code. In this case `window.onerror` will not be called.

**Comment by typhonrt** [comment]
As mentioned on Discord there is no practical reason to do anything with caught errors / handled rejections in someone's code as that is intentional control flow. What makes sense is to capture / record uncaught errors / unhandled rejections and then intuit the source of the error to Foundry or specific modules / systems by filtering the stack trace. The above code snipped should work vs `onerror`. You receive the same errors that subsequently get posted to the console.

**Comment by akrigline** [comment]
@IamWarHead 

I agree that handled errors don't need to be included in this initially. If we want to support that we would have to provide some logging function to module devs like you've suggested in the past, but that's an alright short-fall right now.

Using the eventListeners for unhandledrejections and errors is already going to be useful and will provide a proof of concept for more functionality later. That's a great lead, thanks @typhonrt

**Comment by JiDW** [comment]
+1 for this
Since we can't easily ask the user to check their console for errors as they may not follow the issue created, some reports can never be investigated.
